syntax = "proto3";
package project;




import "model/pipeline/project.proto";
import "model/pipeline/provider.proto";
import "model/pipeline/pipeline.proto";

/*
List请求
*/
message ListRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 page_size = 2 ;
}

/*
List返回
*/
message ListResponse {
message List {
    /*
    代码项目
    */
    pipeline.Project project = 1 ;
    /*
    project关联的provider信息
    */
    pipeline.Provider provider = 2 ;
    /*
    project关联的pipeline信息
    */
    repeated pipeline.Pipeline pipeline = 3 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListApi返回
*/
message ListResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    project.ListResponse data = 4 ;
}

